atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/sweep/AbstractSweepTaskRunnerTest.java

```diff
@@ -513,6 +524,35 @@ public void testSweepBatches() {
         }
         // The last batch can be smaller than deleteBatchSize
         assertThat(sweptCells.get(sweptCells.size() - 1).size()).isLessThanOrEqualTo(2 * deleteBatchSize);
+
+        assertEquals("Expected Ts Pairs Examined should add up to entire table (2 values in each cell)",
+                2 * BIG_LIST_OF_CELLS.size(), sweepResults.getCellTsPairsExamined());
+    }
+
+    @Test(timeout = 50000)
+    public void testSweepBatchesInDifferentRows() {
+        CellsSweeper cellsSweeper = Mockito.mock(CellsSweeper.class);
+        SweepTaskRunner spiedSweepRunner =
+                new SweepTaskRunner(kvs, tsSupplier, tsSupplier, txService, ssm, cellsSweeper);
+
+        putTwoValuesInEachCell(BIG_LIST_OF_CELLS_IN_DIFFERENT_ROWS);
+
+        int deleteBatchSize = 2;
+        Pair<List<List<Cell>>, SweepResults> sweptCellsAndSweepResults = runSweep(cellsSweeper, spiedSweepRunner,
+                10, 1, deleteBatchSize);
+        List<List<Cell>> sweptCells = sweptCellsAndSweepResults.getLhSide();
+        SweepResults sweepResults = sweptCellsAndSweepResults.getRhSide();
+        assertThat(Iterables.concat(sweptCells)).containsExactlyElementsOf(BIG_LIST_OF_CELLS_IN_DIFFERENT_ROWS);
+        for (List<Cell> sweptBatch : sweptCells.subList(0, sweptCells.size() - 1)) {
+            // We requested deleteBatchSize = 2, so we expect between 2 and 4 timestamps deleted at a time.
```

Comment:
nit: if I'm reading it correctly, you want start-inclusive but end-exclusive, since each row has at most 2 (while this is both start and end inclusive).

Reply:
@jeremyk-91 What do you mean by 'this' in this case? Do you mean that the swept batch should have between 2 and 3 timestamps deleted?

Reply:
@gsheasby that's what I meant - but disregard. I think I got confused, thinking that we had a max-cell-ts-pairs of 2 (it is 1000).